src/misc/ansi_code_markup.cpp

```diff
@@ -25,97 +25,213 @@
 #include "support.h"
 #include "ansi_code_markup.h"
 
-class ColorParser {
+/*!
+ * \brief Represents a markup tag.
+ *
+ * The constructor takes the string of a tag name and its optional value, and
+ * attempts to construct a valid tag that can be easily queried for its
+ * properties.
+ *
+ * The tag will be valid if, and only if, valid() returns true. Otherwise the
+ * tag could not be recognized.
+ */
+class Tag {
 public:
+	enum class TagGroup {
+		InvalidGroup,
+		Colors,
+		Erasers,
+		Styles,
+		Misc,
+	};
+
+	enum class TagName {
+		InvalidName,
```

Comment:
Presumably, everything inside the Tag class is tag-related; if possible - we could make the sub-items leaner in their names.  I found myself starting to get tag-fatigue, and my eyes were soon skimming past/ignoring the **Tag*** prefix:

![2022-01-12_12-52](https://user-images.githubusercontent.com/1557255/149220012-c93504ec-c983-4cd9-8c1a-0c22903e2531.png)

and


![2022-01-12_12-51](https://user-images.githubusercontent.com/1557255/149220015-0b2da636-64bf-4f01-a351-fd470a626773.png)

Reply:
Oof, yeah, see what you mean.

Consequence of moving the enums inside the class. If you're happy to keep them there, I'll remove the `Tag` prefix from them.